cmake/CMakeLists.txt

```diff
@@ -23,6 +23,47 @@ else (MSVC)
   set(protobuf_WITH_ZLIB_DEFAULT ON)
 endif (MSVC)
 option(protobuf_WITH_ZLIB "Build with zlib support" ${protobuf_WITH_ZLIB_DEFAULT})
+
+option(protobuf_WITHOUT_CPP_COMPILER "Build with C++ compiler support" ON)
```

Comment:
It's more natural to use "protobuf_WITH_CPP_COMPILER" as the option name. You can keep using GOOGLE_PROTOBUF_NO_CPP as the macro name.